zetta_utils/db_annotations/precomp_annotations.py

```diff
@@ -508,25 +550,91 @@ def write_annotations(self, annotations: Sequence[LineAnnotation], all_levels: b
             level_key = f"spatial{level}"
             level_dir = path_join(self.path, level_key)
             if is_local_filesystem(self.path):
-                os.makedirs(level_dir)
-            for x, y, z in product(
-                range(grid_shape[0]), range(grid_shape[1]), range(grid_shape[2])
-            ):
-                chunk_start = self.index.start + Vec3D(x, y, z) * chunk_size
-                chunk_end = chunk_start + chunk_size
-                chunk_bounds = VolumetricIndex.from_coords(
-                    chunk_start, chunk_end, self.index.resolution
+                os.makedirs(level_dir, exist_ok=True)
+
+            # Yes, there are terser ways to do this 3D iteration in Python,
+            # but they result in having to filter the full set of annotations
+            # for every chunk, which turns out to be very slow.  Much faster
+            # is to do one level at a time, filtering at each level, so that
+            # the final filters don't have to wade through as much data.
+            for x in range(grid_shape[0]):
+                print(f"x = {x} of {grid_shape[0]}", end="", flush=True)
+                split_by_x = VolumetricIndex.from_coords(
+                    (
+                        self.index.start[0] + x * chunk_size[0],
+                        self.index.start[1],
+                        self.index.start[2],
+                    ),
+                    (
+                        self.index.start[0] + (x + 1) * chunk_size[0],
+                        self.index.stop[1],
+                        self.index.stop[2],
+                    ),
+                    self.index.resolution,
                 )
                 # pylint: disable=cell-var-from-loop
-                chunk_data: list[LineAnnotation] = list(
-                    filter(lambda d: d.in_bounds(chunk_bounds), annotations)
+                split_data_by_x: list[LineAnnotation] = list(
+                    filter(lambda d: d.in_bounds(split_by_x), annotations)
                 )
-                if not chunk_data:
+                print(f":  {len(split_data_by_x)} lines")
```

Comment:
`logger.debug`, if worth keeping around

Reply:
It's definitely worth keeping — when this operation runs, it can take several hours, and it's very disconcerting (and inconvenient) to have no feedback during that time.  The current prints provide a nicer UX than logger, because I can control the carriage return.

BUT, I get it.  Logger is our standard way of corralling output, and works better when the code is called as part of a subchunkable flow.  So I'll change them all to logger.debug as suggested.